lib/widgets/share.dart

```diff
@@ -0,0 +1,193 @@
+      showErrorDialog(
+        context: context,
+        title: zulipLocalizations.errorSharingTitle,
+        message: zulipLocalizations.errorSharingAccountNotLoggedIn);
+      return;
+    }
+
+    unawaited(navigator.push(
+      SharePage.buildRoute(
+        accountId: initialAccountId,
+        sharedFiles: intentSendEvent.extraStream?.map((sharedFile) {
+          return FileToUpload(
+            content: Stream.value(sharedFile.bytes),
+            length: sharedFile.bytes.length,
+            filename: sharedFile.name,
+            mimeType: sharedFile.mimeType);
+        }),
+        sharedText: intentSendEvent.extraText)));
+  }
+}
+
+class SharePage extends StatelessWidget {
+  const SharePage({
+    super.key,
+    required this.sharedFiles,
+    required this.sharedText,
+  });
+
+  final Iterable<FileToUpload>? sharedFiles;
+  final String? sharedText;
+
+  static AccountRoute<void> buildRoute({
+    required int accountId,
+    required Iterable<FileToUpload>? sharedFiles,
+    required String? sharedText,
+  }) {
+    return MaterialAccountWidgetRoute(
+      accountId: accountId,
+      page: SharePage(
+        sharedFiles: sharedFiles,
+        sharedText: sharedText));
+  }
+
+  void _handleNarrowSelect(BuildContext context, Narrow narrow) {
+    final messageListPageStateKey = GlobalKey<MessageListPageState>();
+
+    // Push the message list page, replacing the share page.
+    unawaited(Navigator.pushReplacement(context,
+      MessageListPage.buildRoute(
+        key: messageListPageStateKey,
+        context: context,
+        narrow: narrow)));
+
+    // Wait for the message list page to accommodate in the widget tree from the route.
+    SchedulerBinding.instance.addPostFrameCallback((_) async {
+      final messageListPageState = messageListPageStateKey.currentState;
+      if (messageListPageState == null) return; // TODO(log)
+      final composeBoxState = messageListPageState.composeBoxState;
+      if (composeBoxState == null) return; // TODO(log)
+
+      final composeBoxController = composeBoxState.controller;
+
+      // We can receive both: the file/s and an accompanying text,
+      // so first populate the compose box with the text, if there is any.
+      if (sharedText case var text?) {
+        if (!text.endsWith('\n')) text += '\n';
+
+        // If there are any shared files, add a separator new line.
+        if (sharedFiles != null) text += '\n';
+
+        // Populate the text.
+        final contentController = composeBoxController.content;
+        contentController.value =
+          contentController.value
+            .replaced(contentController.insertionIndex(), text);
+      }
+      // Then upload the files and populate the compose box with their links.
+      if (sharedFiles != null) {
+        await composeBoxState.uploadFiles(sharedFiles!);
+      }
+
+      // Try to focus on the topic compose box if there is one,
+      // else focus on content compose box, if not already focused.
```

Comment:
I think specifically that means this latter step should happen before doing an `await` on the former step.

Reply:
Thanks! Fixed in current revision, and also updated videos, PTAL.